File: day6/palindromeCheck.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    cout << "Please enter a word or phrase: ";
    string userInput;
    getline(cin, userInput);

    string cleaned = "";
    for (char c : userInput) {
        if (c != ' ') {
            cleaned += tolower(c);
        }
    }

    string reversed = cleaned;
    reverse(reversed.begin(), reversed.end());

    if (cleaned == reversed) {
        cout << "This is a palindrome." << endl;
    } else {
        cout << "This is not a palindrome." << endl;
    }

    return 0;
}

File: day6/PalindromeCheck.java
import java.util.Scanner;

public class PalindromeCheck {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Please enter a word or phrase: ");
        String userInput = scanner.nextLine();

        String cleaned = userInput.replaceAll(" ", "").toLowerCase();
        String reversed = new StringBuilder(cleaned).reverse().toString();

        if (cleaned.equals(reversed)) {
            System.out.println("This is a palindrome.");
        } else {
            System.out.println("This is not a palindrome.");
        }

        scanner.close();
    }
}
